cxx/irm_test.cc

```diff
@@ -0,0 +1,41 @@
+// Apache License, Version 2.0, refer to LICENSE.txt
+
+#define BOOST_TEST_MODULE test IRM
+
+#include "irm.hh"
+#include "util_distribution_variant.hh"
+
+#include <boost/test/included/unit_test.hpp>
+namespace tt = boost::test_tools;
+
+BOOST_AUTO_TEST_CASE(test_irm) {
```

Comment:
Could you add a test for logp? Maybe something like this (assuming I understand correctly):

```
double logp_score0 = irm.logp_score();
double logp1 = irm.logp(x);
irm.incorporate(x);
double logp_score1 = irm.logp_score();
BOOST_TEST(logp_score1 == logp_score0 + logp1);
```
(Or fine to do in a follow-up, if you or Srinivas are planning on expanding test coverage/adding HIRM tests later.)

Reply:
Well, I added that test, which I also expected to pass, but it didn't.  Added a TODO for now.